### PR TITLE
Update Button Component

### PR DESCRIPTION
**Pull request by codejagaban**
This PR updates the button component  to use icons and icon positionings

### CODE REVIEW

File: src/components/elements/Button.astro [hunk 1]
```diff
@@ -5,9 +5,11 @@ interface Props {
   style: "primary" | "secondary";
   size: "md" | "lg";
   fullWidth?: boolean;
+  icon?: string;
+  iconPosition?: "left" | "right";
 }
 
-const { href, label, style, size, fullWidth } = Astro.props;
+const { href, label, style, size, fullWidth, icon, iconPosition = "left" } = Astro.props;
```
Comment (CodeLeom) [inline src/components/elements/Button.astro:12]: do we need to declare a value for the icon position here?

**Review by CodeLeom — COMMENTED** [review]
this looks ok. Check the comment on the destructuring part

File: src/components/elements/Button.astro (hunk 1 above)
Comment (codejagaban) [reply src/components/elements/Button.astro:12]: This is a default value, in a case where an icon is present and no position is specified, it takes the default value here (left)

**Comment by codejagaban** [comment]
> this looks ok. Check the comment on the destructuring part

@CodeLeom  please approve

**Review by CodeLeom — APPROVED** [review]
:check:

**Pull request merged**